controllers/cluster_controller.go

```diff
@@ -699,6 +699,44 @@ func (r *FoundationDBClusterReconciler) hasFullReplication(cluster *fdbtypes.Fou
 	return status.Cluster.FullReplication, nil
 }
 
+func (r *FoundationDBClusterReconciler) hasDesiredFaultTolerance(cluster *fdbtypes.FoundationDBCluster) (bool, error) {
+	version, err := fdbtypes.ParseFdbVersion(cluster.Spec.Version)
+	if err != nil {
+		return false, err
+	}
+
+	// TODO (johscheuer): https://github.com/FoundationDB/fdb-kubernetes-operator/issues/743
+	// For FDB versions lower than 6.2.0 we just approve the replacement.
+	// In 6.1 we only have machine information.
+	if !version.IsAtLeast(fdbtypes.FdbVersion{Major: 6, Minor: 2, Patch: 0}) {
+		return true, nil
+	}
+
+	adminClient, err := r.DatabaseClientProvider.GetAdminClient(cluster, r)
+	if err != nil {
+		return false, err
+	}
+	defer adminClient.Close()
+
+	status, err := adminClient.GetStatus()
+	if err != nil {
+		return false, err
+	}
+
+	expectedFaultTolerance := cluster.DesiredFaultTolerance()
+	log.V(0).Info("Has desired fault tolerance",
+		"namespace", cluster.Namespace,
+		"cluster", cluster.Name,
+		"expectedFaultTolerance", expectedFaultTolerance,
+		"maxZoneFailuresWithoutLosingData", status.Cluster.FaultTolerance.MaxZoneFailuresWithoutLosingData,
+		"maxZoneFailuresWithoutLosingAvailability", status.Cluster.FaultTolerance.MaxZoneFailuresWithoutLosingAvailability)
+
+	return internal.HasDesiredFaultTolerance(
```

Comment:
What's the advantage of looking at two different fields here, rather than relying on `cluster.full_replication`?

Reply:
I believe (I'll dig into the code to confirm this) is that the `cluster.full_replication` is only looking at the data but is not including the availability part (I'm not too sure about that). I'll investigate and find out what's the best solution to figure out the clusters fault tolerance status. As a side product I'm hopefully able to clarify that on the FDB doc (they are not totally clear about the different "metrics").